drivers/interrupt_controller/Kconfig.gic

```diff
@@ -0,0 +1,39 @@
+# ARM Generic Interrupt Controller (GIC) configuration
+
+# Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
+# SPDX-License-Identifier: Apache-2.0
+
+if CPU_CORTEX
+
+config GIC
+	bool
+
+config GIC_V1
```

Comment:
Does it make sense to enable e.g. both `GIC_V1` and `GIC_V2` at the same time?

Could put the `GIC_V*` symbols inside a choice otherwise. If a "no GIC" option is needed, then an [optional choice](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#optional-choices) might be handy.

Reply:
> Does it make sense to enable e.g. both `GIC_V1` and `GIC_V2` at the same time?

No, only one should be selected.

> Could put the `GIC_V*` symbols inside a choice otherwise. If a "no GIC" option is needed, then an [optional choice](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#optional-choices) might be handy.

I tried "optional choice", but it seems like choice symbols cannot be "select"-ed (note that the `GIC_V*` symbols are selected by the SoC Kconfig).

```
Error: Aborting due to non-whitelisted Kconfig warning 'warning: the choice symbol GIC_V1 (defined
at drivers/interrupt_controller/Kconfig.gic:15) is selected by the following symbols, but
select/imply has no effect on choice symbols  - SOC_XILINX_ZYNQMP (defined at
soc/arm/xilinx_zynqmp/Kconfig.soc:4)'.
```

I suppose one way to enforce this would be to implement mutual exclusion using `depends on`, but I wanted to avoid that as much as possible because it is kind of hacky (but then, I guess the same could be said about `GIC_VER`).

Is there a better way to handle this issue? or should I just go with the `depends on` approach?

Reply:
> I suppose one way to enforce this would be to implement mutual exclusion using `depends on`, but I wanted to avoid that as much as possible because it is kind of hacky (but then, I guess the same could be said about `GIC_VER`).

Actually, I just tried this and apparently that is not possible either:
```
Dependency loop
===============

GIC_V1 (defined at drivers/interrupt_controller/Kconfig.gic:11), with definition...

config GIC_V1
        bool "ARM Generic Interrupt Controller (GIC) v1"
        select GIC
        depends on !GIC_V2 && !GIC_V3 && CPU_CORTEX
        help
          The ARM Generic Interrupt Controller v1 (e.g. PL390) works with the
          ARM Cortex-family processors.

(select-related dependencies: SOC_XILINX_ZYNQMP && <choice>)

...depends on GIC_V3 (defined at drivers/interrupt_controller/Kconfig.gic:27), with definition...

config GIC_V3
        bool "ARM Generic Interrupt Controller (GIC) v3"
        select GIC
        depends on !GIC_V1 && !GIC_V2 && CPU_CORTEX
        help
          The ARM Generic Interrupt Controller v3 (e.g. GIC-500 and GIC-600)
          works with the ARM Cortex-family processors.

...depends again on GIC_V1 (defined at drivers/interrupt_controller/Kconfig.gic:11)
```

Reply:
Semi-maybe-unrelated, but do the symbols need to be user-configurable by the way?

If not, then just removing the prompts would solve the problem. Then it won't be possible to enable them in `menuconfig`, and a `select` will be required.

Reply:
> Semi-maybe-unrelated, but do the symbols need to be user-configurable by the way?

At the moment, they do not have to be user-configurable in `menuconfig`, but that may become useful/necessary in the future (e.g. if we ever define a "generic" ARM platform that can be `menuconfig`-ed to support custom platforms).

For now, I will remove the prompts.

Reply:
Maybe having them [optionally configurable](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#optional-prompts) could be handy at that point too.